odl/solvers/functional/default_functionals.py

```diff
@@ -570,6 +573,465 @@ def __init__(self, space):
+                    tmp = np.log(x / functional.prior)
+
+                if np.all(np.isfinite(tmp)):
+                    return tmp
+                else:
+                    # The derivative is not defined.
+                    raise ValueError('The gradient of the Kullback-Leibler '
+                                     'Cross Entropy functional is not defined '
+                                     'for `x` with one or more components '
+                                     'less than or equal to zero.'.format(x))
+
+        return KLCrossEntropyGradient()
+
+    @property
+    def proximal(self):
+        """Return the proximal factory of the functional.
+
+        See Also
+        --------
+        proximal_cconj_kl_cross_entropy : proximal factory for convex conjugate
+        of KL cross entropy.
+        proximal_cconj : proximal of the convex conjugate of a functional
+        """
+        return proximal_cconj(proximal_cconj_kl_cross_entropy(
+            space=self.domain, g=self.prior))
+
+    @property
+    def convex_conj(self):
+        """The convex conjugate functional of the KL-functional."""
+        return KullbackLeiblerCrossEntropyConvexConj(self.domain, self.prior)
+
+
+class KullbackLeiblerCrossEntropyConvexConj(Functional):
+    """The convex conjugate of Kullback-Leibler Cross Entorpy functional.
+
+    Notes
+    -----
+    The functional is given by
+
+    .. math::
+        \\sum_i g_i (exp(x_i) - 1)
+
+    where :math:`g` is the prior.
+
+    This is the convex conjugate functional of `KullbackLeiblerCrossEntropy`.
+
+    See Also
+    --------
+    KullbackLeiblerCrossEntropy : convex conjugate functional
+    """
+
+    def __init__(self, space, prior=None):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        space : `DiscreteLp` or `FnBase`
+            Domain of the functional.
+        g : ``space`` element, optional
+            Data term, positive.
+            Default: if None it is take as the one-element.
+        """
+        super().__init__(space=space, linear=False, grad_lipschitz=np.nan)
+
+        if prior is not None and prior not in self.domain:
+            raise ValueError('`prior` not in `domain`'
+                             ''.format(prior, self.domain))
+
+        self.__prior = prior
+
+    @property
+    def prior(self):
+        """The prior in convex conjugate Kullback-Leibler Cross Entorpy."""
+        return self.__prior
+
+    # TODO: update when integration operator is in place: issue #440
+    def _call(self, x):
+        """Return the value in the point ``x``."""
+        if self.prior is None:
+            tmp = (np.exp(x) - 1).inner(self.domain.one())
+        else:
+            tmp = (self.prior * (np.exp(x) - 1)).inner(self.domain.one())
+        return tmp
+
+    @property
+    def gradient(self):
+        """Gradient operator of the functional."""
+        functional = self
+
+        class KLCrossEntCCGradient(Operator):
```

Comment:
Add a todo of replacing this once the `UFuncOperators` come in. Also that branch needs a review :/